src/experimental/patterns/ec2-app.ts

```diff
@@ -0,0 +1,307 @@
+}
+
+/**
+ * An IAM Policy allowing the sending of a CloudFormation signal.
+ *
+ * @privateRemarks
+ * Implemented as a singleton as the resources can only be tightened at most to the CloudFormation stack.
+ */
+class AsgRollingUpdatePolicy extends Policy {
+  private static instance: AsgRollingUpdatePolicy | undefined;
+
+  private constructor(scope: GuStack) {
+    const { stackId } = scope;
+
+    super(scope, "AsgRollingUpdatePolicy", {
+      statements: [
+        // Allow usage of command `cfn-signal`.
+        new PolicyStatement({
+          actions: ["cloudformation:SignalResource"],
+          effect: Effect.ALLOW,
+          resources: [stackId],
+        }),
+
+        /*
+        Allow usage of command `aws elbv2 describe-target-health`.
+        AWS Elastic Load Balancing does not support resource based policies, so the resource has to be `*` (any) here.
+        See https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_aws-services-that-work-with-iam.html
+         */
+        new PolicyStatement({
+          actions: ["elasticloadbalancing:DescribeTargetHealth"],
+          effect: Effect.ALLOW,
+          resources: ["*"],
+        }),
+      ],
+    });
+  }
+
+  public static getInstance(stack: GuStack): AsgRollingUpdatePolicy {
+    if (!this.instance || !isSingletonPresentInStack(stack, this.instance)) {
+      this.instance = new AsgRollingUpdatePolicy(stack);
+    }
+
+    return this.instance;
+  }
+}
+
+export interface GuEc2AppExperimentalProps extends Omit<GuEc2AppProps, "updatePolicy"> {}
+
+/**
+ * An experimental pattern to instantiate an EC2 application that is updated entirely via CloudFormation.
+ * It sets the update policy of the `AWS::AutoScaling::AutoScalingGroup` to `AutoScalingRollingUpdate`.
+ * When a CloudFormation update is applied, the current instances in the ASG will be replaced.
+ *
+ * This pattern also updates the User Data, running some commands AFTER yours.
+ * These changes are wrapped in start and end marking comments.
+ *
+ * This pattern does not enable for scaling events to trigger _during_ a deployment
+ * as during testing it was found that scale-in events during deployment led to unpredictable service availability.
+ *
+ * If a deployment fails, for example when the healthcheck fails, CloudFormation will automatically rollback the changes.
+ * This improves on the behaviour of Riff-Raff's `autoscaling` deployment which requires manual intervention on deployment failure.
+ *
+ * To update the application's code, a CloudFormation update must be triggered.
+ * The best way to do this is to include the build number in the application artifact.
+ * TODO test User Data includes a build number.
+ *
+ * @remarks
+ * - The rate at which instances are replaced are conditional based on how scaled up the service is.
+ *   The more an application is scaled, the longer it'll take to deploy.
+ * - It is not necessary to use the `autoscaling` Riff-Raff deployment type in conjunction with this.
```

Comment:
https://github.com/guardian/security-hq/pull/1154 shows the `riff-raff.yaml` changes required if creating the file manually:
1. Use the `autoscaling` deployment type, but only the `uploadArtifacts` action
2. Have the `cloud-formation` deployment type depend on step 1

Obviously, it's not required, but a reduced risk migration path would look like this:
1. Update to latest version of GuCDK
2. Migrate to using the auto-generated `riff-raff.yaml` file
3. Use the experimental pattern

Where step 1 and 2 can go to `main` independent of 3.

Reply:
I'll add this nuanced detail to the release notes.

Reply:
Done - https://github.com/guardian/cdk/pull/2417/commits/f3072e101d71bba24ba987b416b6753d1c1c4834.